### PR TITLE
Send an X-App-ID: FireCloud http header with every* ajax request

### PR DESCRIPTION
**Pull request by davidangb**
requires broadinstitute/firecloud-develop#1307 (which is now merged)

This adds an X-App-ID: FireCloud http header on all ajax requests to orch, leo, sam, bond, and martha as part of an effort to understand which apps are making requests to our services.

Because this additional header requires CORS configuration, we are not yet sending this header on ajax requests to:
* Dockstore
* Google
* anything other than orch/leo/sam/bond/martha

we don't care about the first two unless they ask us for it. I am not aware of any other ajax requests to Workbench services that are not orch/leo/sam/bond/martha.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/utils/ajax.cljs [hunk 1]
```diff
@@ -12,6 +12,8 @@
 ;; TODO - make this unnecessary
 (def content-type=json {"Content-Type" "application/json"})
 
+(def app-id {"X-App-Id" "FireCloud"})
+
```
Comment (ghost) [inline src/cljs/main/broadfcui/utils/ajax.cljs]: I noticed that the header keys for the various services in [firecloud-develop/1307](https://github.com/broadinstitute/firecloud-develop/pull/1307) were either all lower case or kabob/camel case. Is it safe to assume the header is not case sensitive?

**Review by ghost — APPROVED** [review]
One question inline. Looks good to me!

File: src/cljs/main/broadfcui/utils/ajax.cljs (hunk 1 above)
Comment (davidangb) [reply src/cljs/main/broadfcui/utils/ajax.cljs]: pretty much everything on the internet - plus empirical testing - says these comparisons are not case-sensitive.

**Comment by davidangb** [comment]
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8561/
```
[info] *** 2 TESTS FAILED ***
[error] Failed: Total 124, Failed 2, Errors 0, Passed 122, Ignored 5
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainMultiGroupsSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```

**Comment by davidangb** [comment]
`AuthDomainMultiGroupsSpec` passed in a manual run: https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8562/

**Review by ghost — APPROVED** [review]

**Pull request merged**